src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_CollectionExpression.cs

```diff
@@ -299,28 +309,28 @@ private BoundExpression VisitCollectionBuilderCollectionExpression(BoundCollecti
             return result;
         }
 
-        internal static bool ShouldUseRuntimeHelpersCreateSpan(BoundCollectionExpression node, TypeSymbol elementType)
+        internal static bool IsAllocatingRefStructCollectionExpression(BoundCollectionExpressionBase node, TypeSymbol targetType, TypeSymbol? elementType, CSharpCompilation compilation)
         {
-            Debug.Assert(node.CollectionTypeKind is
-                CollectionExpressionTypeKind.ReadOnlySpan or
-                CollectionExpressionTypeKind.CollectionBuilder);
+            return targetType.IsRefLikeType
+                && !node.Elements.IsEmpty
```

Comment:
>!node.Elements.IsEmpty [](http://example.com/codeflow?start=19&length=22)

Perhaps use `node.Elements.Length > 0` for consistency with helper methods below.